packages/shared/scope-hoisting/src/link.js

```diff
@@ -212,9 +212,7 @@ export function link({
         );
         let parent;
         if (binding) {
-          parent = path.findParent(
-            p => getScopeBefore(p) === binding.scope && p.isStatement(),
-          );
+          parent = path.findParent(p => t.isProgram(p.parent));
```

Comment:
Not sure if Will forwarded this to you, but can you please add a invariant here to make sure that `binding.path.parentPath.isProgram()` is true to prevent a broken bundle output in the future.